src/main/java/com/xebialabs/deployit/ci/Credential.java

```diff
@@ -280,7 +281,9 @@ public ListBoxModel doFillCredentialsIdItems(@AncestorInPath Project context) {
             return new StandardUsernameListBoxModel().withAll(creds);
```

Comment:
This method needs a permission check. See credentials plugin docs for how it should look like.

Reply:
added permission check.
```
if (context == null && !Jenkins.getActiveInstance().hasPermission(Jenkins.ADMINISTER) ||
        context != null && !context.hasPermission(context.EXTENDED_READ) &&
                !context.hasPermission(CredentialsProvider.USE_ITEM)) {
    return new StandardUsernameListBoxModel();
}
```